### ISSUE TITLE
Add Sphere3d type and module

### ISSUE THREAD

**Issue by ianmackenzie** [opensolid/geometry#11]
`type Sphere3d = Sphere3d { centerPoint : Point3d, radius : Float }`

This would be very similar to the `Circle2d` type and module, so that could be used as an example/template. A couple of interesting `Sphere3d`-specific additions:

```elm
projectOnto : Plane3d -> Sphere3d -> Circle3d
projectInto : SketchPlane3d -> Sphere3d -> Circle2d
```

**Comment by daphee** [comment]
What exactly would projectOnto do? I guess something like

```
projectOnto : Plane3d -> Sphere3d -> Circle3d
projectOnto plane sphere =
        Circle3d.with 
             { centerPoint = Point3d.projectOnto plane (centerPoint sphere)
             , axialDirection = Plane3d.normalDirection plane
             , radius = ???
             }
```

'm not sure about how the radius would transform.

**Comment by ianmackenzie** [comment]
Yes, exactly, and `radius = Sphere3d.radius sphere` since this is just a simple orthographic projection, not something like a perspective projection which would distort things. Do you think the docs should make that more clear?

**Comment by daphee** [comment]
I think "orthographically" is better than perpendicularly, mainly because the Wikipedia page on "orthographical projection" has a nice immediately visible picture which can make clear what the projection does. 😃 

I'll need to write some more tests and will create a PR in the next few days. 

One more question: are you using elm-format? if yes, do you have some customized settings? The reason I ask is that I have my editor set to automatically apply elm-format and now in the files I edited the bodies of all `let` ... `in` ... expressions are shifted to the right by 4 spaces.

**Comment by daphee** [comment]
>  I'm not sure it extends to projection onto axes, though - would you keep "perpendicular" for that?

Hmm... Not sure on that one. On the one hand it is nice to have everything consistent. On the other hand "perpendicular" is the more common / ""easier"" word. 

> In the future I'll try to be more careful to use the stable/standard version of elm-format, but in the meantime if you could use 0.7.0-exp for OpenSolid stuff that would be great. Thanks, and sorry for the annoyance!

Cool, I'll do that. I'm not doing anything else with Elm at the moment so it shouldn't be any problem to switch to the 0.7.0-exp version.

**Comment by ianmackenzie** [comment]
Implemented in #32 - thanks @daphee!
